src/go/package/app/models/package_test.go

```diff
@@ -29,6 +29,168 @@ func (s *PackageSuite) SetupTest() {
+					}).Once()
+				mockOrmQuery.On("Association", "Tags").Return(mockOrmAssociation).Once()
+				mockOrmAssociation.On("Replace", mock.MatchedBy(func(tags []*Tag) bool {
+					return len(tags) == 1 && tags[0].Name == "goravel"
+				})).Return(nil).Once()
+			},
+		},
+		{
+			name: "Sad path - Tags association error",
+			setup: func() {
+				beforeSetup()
+				mockOrm.On("Query").Return(mockOrmQuery).Once()
+				mockOrmQuery.On("WhereIn", "name", []any{"goravel"}).Return(mockOrmQuery).Once()
+				mockOrmQuery.On("Find", mock.AnythingOfType("*[]*models.Tag")).
+					Return(nil).
+					Run(func(args mock.Arguments) {
+						tagPtr := args.Get(0).(*[]*Tag)
+						*tagPtr = []*Tag{{UUIDModel: UUIDModel{ID: 1}, Name: "goravel", UserID: userID}}
+					}).Once()
+
+				mockOrmQuery.On("Model", mock.AnythingOfType("*models.Package")).Return(mockOrmQuery).
+					Run(func(args mock.Arguments) {
+						packagePtr := args.Get(0).(*Package)
+						packagePtr.Tags = tagModels
+					}).Once()
+				mockOrmQuery.On("Association", "Tags").Return(mockOrmAssociation).Once()
+				mockOrmAssociation.On("Replace", mock.MatchedBy(func(tags []*Tag) bool {
+					return len(tags) == 1 && tags[0].Name == "goravel"
+				})).Return(errors.New("error")).Once()
+			},
+			expectedErr: utilserrors.New(http.StatusInternalServerError, "error"),
+		},
+		{
+			name: "Sad path - Find tag error",
+			setup: func() {
+				beforeSetup()
+				mockOrmQuery.On("WhereIn", "name", []any{"goravel"}).Return(mockOrmQuery).Once()
+				mockOrmQuery.On("Find", mock.AnythingOfType("*[]*models.Tag")).
+					Return(errors.New("error")).Once()
+			},
+			expectedErr: utilserrors.New(http.StatusInternalServerError, "error"),
+		},
+		{
+			name: "Happy path - Tags do not exist",
+			setup: func() {
+				beforeSetup()
+				mockOrm.On("Query").Return(mockOrmQuery).Twice()
+				mockOrmQuery.On("WhereIn", "name", []any{"goravel"}).Return(mockOrmQuery).Once()
+				mockOrmQuery.On("Find", mock.AnythingOfType("*[]*models.Tag")).
+					Return(nil).Once()
+
+				mockOrmQuery.On("Create", mock.MatchedBy(func(tags []*Tag) bool {
+					return len(tags) == 1 && tags[0].Name == "goravel" && tags[0].UserID == userID && tags[0].IsShow == 1
+				})).Return(nil).
+					Run(func(args mock.Arguments) {
+						tagPtr := args.Get(0).([]*Tag)
+						tagPtr[0] = &Tag{UUIDModel: UUIDModel{ID: 1}, Name: "goravel", UserID: userID}
+					}).Once()
+
+				mockOrmQuery.On("Model", mock.AnythingOfType("*models.Package")).Return(mockOrmQuery).
+					Run(func(args mock.Arguments) {
+						packagePtr := args.Get(0).(*Package)
+						packagePtr.Tags = tagModels
+					}).Once()
+				mockOrmQuery.On("Association", "Tags").Return(mockOrmAssociation).Once()
+				mockOrmAssociation.On("Replace", mock.MatchedBy(func(tags []*Tag) bool {
+					return len(tags) == 1 && tags[0].Name == "goravel"
+				})).Return(nil).Once()
+			},
+		},
+		{
+			name: "Sad path - Create Tag error",
+			setup: func() {
+				beforeSetup()
```

Comment:
Why add this here? It exists in the for logic below.

Reply:
sorry by mistake